src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -152,17 +152,17 @@ private X509Certificate ServerCertSelectionCallbackWrapper(string targetHost)
 
         private SslAuthenticationOptions CreateAuthenticationOptions(SslServerAuthenticationOptions sslServerAuthenticationOptions)
         {
-            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null)
+            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null && _certSelectionDelegate == null)
             {
                 throw new ArgumentNullException(nameof(sslServerAuthenticationOptions.ServerCertificate));
             }
 
-            if (sslServerAuthenticationOptions.ServerCertificate != null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback != null)
+            if (sslServerAuthenticationOptions.ServerCertificate != null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback != null  && _certSelectionDelegate == null)
             {
                 throw new InvalidOperationException(SR.Format(SR.net_conflicting_options, nameof(ServerCertificateSelectionCallback)));
             }
 
-            var authOptions = new SslAuthenticationOptions(sslServerAuthenticationOptions);
+            var authOptions = new SslAuthenticationOptions(sslServerAuthenticationOptions, _certSelectionDelegate);
```

Comment:
I'd assign it directly as we do below for ServerCertSelectionDelegate

Reply:
Assigning it directly makes things really messy, since both of these lamdas would go into the using statement. I think that this way keeps things cleaner.

Reply:
(I mean to not create new arg and instead assign property/field)

Reply:
My bad -- I thought this was at another point in the code :)

Reply:
If this method wasn't constructing SslAuthenticationOptions instance I'd agree :) logic in this class is slightly convoluted so I understand the confusion 😄

Reply:
And we should set it only if `sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions.ServerCertificateSelectionCallback == null`, otherwise we can dump the value here (we should add comment that it is compat-only for using old-style .ctor overload which worked also on Server